File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        ArrayList<Character> one = new ArrayList<>();
        ArrayList<Character> two = new ArrayList<>();
        for(int i=0; i<s.length(); i++){
            char ch = s.charAt(i);
            one.add(ch);
        }
        for(int i=0; i<t.length(); i++){
            char ch = t.charAt(i);
            two.add(ch);

        }
        Collections.sort(one);
        Collections.sort(two);
        if(one.equals(two)){
            return true;
        }
        return false;
    }
}

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        freq = {}
        for ch in s:
            freq[ch] = freq.get(ch, 0)+1
        for ch in t:
            if ch not in freq or freq[ch] == 0:
                return False
            freq[ch] = freq.get(ch, 0)-1

        for v in freq.values():
            if v>0:
                return False

        return True